### PR TITLE
New rule S6798: [JSInvokable] attribute should only be used on public methods

### LINKED ISSUE #8278: New rule S6798: [JSInvokable] attribute should only be used on public methods
RSPEC Preview: https://sonarsource.github.io/rspec/#/rspec/S6798
RSPEC PR: https://github.com/SonarSource/rspec/pull/3210

### PR DESCRIPTION
**Pull request by antonioaversa**
Fixes #8278

### CODE REVIEW

**Comment by antonioaversa** [comment]
@sebastien-marichal Assigning this PR to you despite UTs failing since the UTs failures seem to all come from the issue with xunit that we also have on the master branch.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/JSInvokableMethodsShouldBePublic.cs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    using AliasForJSInterop = Microsoft.JSInterop;
+    using AliasForJSInvokable = Microsoft.JSInterop.JSInvokableAttribute;
+
+    class Test
+    {
+        [JSInvokable] private void WithoutAttributeSuffix() { }                     // Noncompliant
+        [JSInvokableAttribute] private void WithAttributeSuffix() { }               // Noncompliant
+        [Microsoft.JSInterop.JSInvokable] private void WithFullyQualifiedName() { } // Noncompliant
+        [AliasForJSInterop.JSInvokable] private void WithNamespaceAlias() { }       // Noncompliant
+        [AliasForJSInvokable] private void WithAttributeAlias() { }                 // FN
```
Comment (sebastien-marichal) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/JSInvokableMethodsShouldBePublic.cs:111]: I would add a bit of context as to why this FN is not fixed.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/DiagnosticVerifier.cs [hunk 2]
```diff
@@ -245,7 +245,7 @@ private static void VerifyBuildErrors(ImmutableArray<Diagnostic> diagnostics, Co
             var buildErrors = GetBuildErrors(diagnostics).ToArray();
 
             var expectedBuildErrors = compilation.SyntaxTrees
-                                                 .Skip(1)
+                                                 .Where(x => !x.FilePath.Contains("ExtraEmptyFile"))
```
Comment (sebastien-marichal) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/DiagnosticVerifier.cs:248]: You should be able to use the `ExceptExtraEmptyFile` extension method.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/Razor/EmptyProject/_Imports.razor [hunk 3]
```diff
@@ -4,5 +4,4 @@
 @using Microsoft.AspNetCore.Components.Forms
 @using Microsoft.AspNetCore.Components.Routing
 @using Microsoft.AspNetCore.Components.Web
-@using Microsoft.AspNetCore.Components.Web.Virtualization
-@using Microsoft.JSInterop
\ No newline at end of file
+@using Microsoft.AspNetCore.Components.Web.Virtualization
```
Comment (sebastien-marichal) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/Razor/EmptyProject/_Imports.razor:7]: Nitpick: add an empty line at the end of the file.

**Review by sebastien-marichal — APPROVED** [review]
LGTM!
I left only nitpick comments.

Nice test cases!

**Pull request merged**